src/Resources/config/gaufrette.php

```diff
@@ -16,6 +16,9 @@
 use Gaufrette\Adapter\Ftp;
 use Gaufrette\Adapter\OpenCloud;
 use Gaufrette\Filesystem;
+use OpenCloud\ObjectSource\Service;
+use OpenCloud\OpenStack;
+use OpenCloud\Rackspace;
```

Comment:
This integration should probably be removed: https://github.com/rackspace/php-opencloud/blob/working/composer.json#L24-L29

Last release 2016... and I don't think it works... it forces the install of https://packagist.org/packages/guzzle/guzzle#v3.8.1

the 3.8.1 , and not the 3.9.1 because on 3.8.1 the symfony/event-dispatcher was >=2.1. On 3.9 they moved to: symfony/event-dispatcher: ~2.1

At the end, it probably does not work. I would prefer leaving this as is, and open another PR to remove this integration. IMO if we decide to not remove it, we shouldn't add it to composer.json, even in require-dev this could cause a lot of trouble, it is a really outdated code and dependencies (guzzle/guzzle is deprecated too)

Probably all the gaufrette could be removed in favour of flysystem, but that will be a lot of work probably... not going to land for 4.0.

Reply:
wdyt? @VincentLanglet

Reply:
More info, this is the dependency: https://github.com/rackspace/php-opencloud

And it is deprecated too, in favour of this one: https://github.com/php-opencloud/openstack

A little bit better but the gauffrete adapter does not work with that one, only with the other.

Reply:
https://github.com/php-opencloud/openstack is recommended in the readme of https://github.com/rackspace/php-opencloud.
So https://github.com/rackspace/php-opencloud can be removed indeed.

Reply:
So, IMO:

1. This PR fixes an issue with having parameters for just class names and also not showing what are our dependencies because adding classes as strings instead of ::class, so it should be merged.
2. Open issue to talk about gauffrete (maybe for 5.0)
3. Open issue to remove OpenStack
4. Deprecated OpenStack on 3.x
5. Remove openStack on master

is that right?